win32ss/gdi/gdi32/objects/text.c

```diff
@@ -18,18 +25,21 @@ TextOutA(
     ANSI_STRING StringA;
     UNICODE_STRING StringU;
     BOOL bResult;
+    INT cchWide;
 
     if (lpString != NULL)
     {
         RtlInitAnsiString(&StringA, (LPSTR)lpString);
         RtlAnsiStringToUnicodeString(&StringU, &StringA, TRUE);
+        cchWide = MultiByteToWideChar(CP_ACP, 0, lpString, cchString, NULL, 0);
     }
     else
     {
         StringU.Buffer = NULL;
+        cchWide = 0;
     }
 
-    bResult = TextOutW(hdc, nXStart, nYStart, StringU.Buffer, cchString);
+    bResult = TextOutW(hdc, nXStart, nYStart, StringU.Buffer, cchWide);
```

Comment:
s/cchWide/StringU.Length/

Reply:
You mean `StringU.Length / sizeof(WCHAR)` :p

Reply:
Good point :) Since these lengths are stored in number of bytes, while here the TextOutW needs a number of chars.